File: fixtures/scale/ruby/large/lib/check_lcgstream.rb
require_relative 'lcgstream_encrypt'
require_relative 'lcgstream_decrypt'
require_relative 'corpus_stream'

def check_lcgstream
  corpus_stream.each do |text|
    sealed = lcgstream_encrypt(text, 8)
    opened = lcgstream_decrypt(sealed, 8)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_lcgstream.py
from cipherlab.ciphers.lcgstream import lcgstream_encrypt, lcgstream_decrypt
from cipherlab.specs.lcgstream_spec import lcgstream_spec
from cipherlab.corpus.corpus_stream import corpus_stream


def check_lcgstream():
    spec = lcgstream_spec()
    for text in corpus_stream():
        sealed = lcgstream_encrypt(text, spec["key"])
        opened = lcgstream_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "stream"
